### PR TITLE
Adding scripts to perform and sign a Maven release.

### PR DESCRIPTION
**Pull request by fnothaft**
Adds on to the initial Mavenization from #115. With this (and a properly configured `~/.m2/settings.xml`) an authorized user can push a release to [Sonatype](http://oss.sonatype.com) by running `./scripts/release/release.sh`.

### CODE REVIEW

File: scripts/release/README [hunk 1]
```diff
@@ -0,0 +1,20 @@
+This document describes how to make an ADAM release.
```
Comment (jmarshall) [inline scripts/release/README]: ADAM?

Comment (fnothaft) [inline scripts/release/README]: Ha, sorry. I copied these scripts from another project that I work on and forgot to change the documentation file. My apologies.

Comment (fnothaft) [inline scripts/release/README]: Fixed.

**Comment by nh13** [comment]
Hey Frank, thank-you for helping with the Maven release scripts as well as pushing us along to do this.  We have a [branch](https://github.com/samtools/htsjdk/tree/jc_sbt_build) on which we made a [Maven release](https://repo1.maven.org/maven2/com/github/samtools/htsjdk/1.128/) using SBT.  We hope to merge it into master soon (after cleanup and code review) and it will be part of our release process to publish to maven central.  My apologies for some duplicated effort but it has really helped having your help and the community call to action to get this one.

**Pull request closed without merging**